fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/expression/ExpressionUtils.java

```diff
@@ -0,0 +1,140 @@
+ */
+public class ExpressionUtils {
+
+    /**
+     * Factory function to create a new {@link PredicateAdapter} using the given {@link Predicate}
+     * as the current expression.
+     * @return
+     */
+    public static PredicateAdapter predicate() {
+        return new PredicateAdapter();
+    }
+
+    public static WhereFragment on(String col) {
+        WhereFragment where = new WhereFragment();
+        where.col(col);
+        return where;
+    }
+
+    public static WhereFragment on(String tableAlias, String col) {
+        WhereFragment where = new WhereFragment();
+        where.col(tableAlias, col);
+        return where;
+    }
+
+    /**
+     * Factory function to create a new {@link Exists} predicate expression
+     * based on the given {@link Select} statement
+     * @param s
+     * @return
+     */
+    public static Exists exists(Select s) {
+        return new Exists(s);
+    }
+
+    /**
+     * Factory function for creating an {@link Alias} from a string
+     *
+     * @param aliasStr
+     * @return
+     */
+    public static Alias alias(String aliasStr) {
+        return new Alias(aliasStr);
+    }
+
+    /**
+     * Factory function for creating a {@link ColumnRef} instance
+     * from a string
+     * @param ref
+     * @return
+     */
+    public static ColumnRef col(String ref) {
+        return new ColumnRef(ref);
+    }
+
+    public static ColumnRef col(String alias, String name) {
+        return new ColumnRef(alias + "." + name);
+    }
+
+    /**
+     * Simple container for a string value to make overloading of methods more natural
+     * @param value
+     * @return
+     */
+    public static LiteralString string(String value) {
+        return new LiteralString(value);
+    }
+
+    /**
+     * Factory function for creating a BindMarkerNode instance
+     * @return
+     */
+    public static BindMarkerNode bind(String value) {
+        return new StringBindMarkerNode(value);
+    }
+
+    public static BindMarkerNode bind(Long value) {
+        return new LongBindMarkerNode(value);
+    }
+
+    public static BindMarkerNode bind(Integer value) {
+        return new IntegerBindMarkerNode(value);
+    }
+
+    public static BindMarkerNode bind(Double value) {
+        return new DoubleBindMarkerNode(value);
+    }
+
+    public static BindMarkerNode bind(Instant value) {
+        return new InstantBindMarkerNode(value);
+    }
+
+    public static BindMarkerNode bind(BigDecimal value) {
+        return new BigDecimalBindMarkerNode(value);
+    }
+
+    public static ExpNode sin(ExpNode arg) {
+        return new SinExpNode(arg);
+    }
+
+    public static ExpNode cos(ExpNode arg) {
+        return new CosExpNode(arg);
+    }
+
+    public static ExpNode acos(ExpNode arg) {
+        return new ACosExpNode(arg);
+    }
+
+    public static ExpNode isDeleted(String alias) {
+        return new WhereFragment().col(alias, "IS_DELETED").eq(string("N")).getExpression();
```

Comment:
use  constants?  also this seems to really blend the jdbc  implementation with database-utils.

Reply:
Constants to be consolidated after whole-system search implementation.